tools/jvm/src/main/scala/org/scalajs/core/tools/linker/backend/closure/ClosureLinkerBackend.scala

```diff
@@ -75,21 +75,51 @@ final class ClosureLinkerBackend(
     module.add(new CompilerInput(ast, ast.getInputId(), false))
 
     // Compile the module
-    val closureExterns =
-      toClosureSource(ClosureLinkerBackend.ScalaJSExternsFile)
+    val closureExterns = List(
+        toClosureSource(ClosureLinkerBackend.ScalaJSExternsFile),
+        toClosureSource(makeExternsForExports(unit)))
     val options = closureOptions(output.name)
     val compiler = closureCompiler(logger)
 
     val result = logger.time("Closure: Compiler pass") {
       compiler.compileModules(
-          List(closureExterns).asJava, List(module).asJava, options)
+          closureExterns.asJava, List(module).asJava, options)
     }
 
     logger.time("Closure: Write result") {
       writeResult(result, compiler, output)
     }
   }
 
+  /** Constructs an externs file listing all exported properties in a linking
+   *  unit.
+   *
+   *  This is necessary to avoid #2491.
```

Comment:
Consider:

This is necessary to avoid name clashes with renamed properties (#2491).

It's not much more text, but will help a potential reader *much* more.